bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/api/VocHttpApi.java

```diff
@@ -61,12 +62,12 @@ public class VocHttpApi {
     private final HttpClient httpClient;
     private final ApiBridgeConfiguration configuration;
 
-    public VocHttpApi(ApiBridgeConfiguration configuration, Gson gson, HttpClient httpClient)
-            throws VolvoOnCallException {
+    public VocHttpApi(String clientName, ApiBridgeConfiguration configuration, Gson gson,
+            HttpClientFactory httpClientFactory) throws VolvoOnCallException {
         this.gson = gson;
         this.cache = new ExpiringCacheMap<>(120 * 1000);
         this.configuration = configuration;
-        this.httpClient = httpClient;
+        this.httpClient = httpClientFactory.createHttpClient(clientName);
```

Comment:
What is the benefit of the changes in this class/constructor? See proposal below.

Reply:
Yes, I hesitated between the two choices.
My final choice was to have `httpClient` fully handled in only one place (class) rather than 2 in the option you propose.